test/testDensity.cpp

```diff
@@ -0,0 +1,288 @@
+	
+	//get<type>Density is independent of type activation, getDensity is not independent 
+	//check if getDensity returns 0. (should give a error message in log file)
+	EXPECT_DOUBLE_EQ(n.getDensity(p), 0);
+	EXPECT_DOUBLE_EQ(n.getNucleonDensity(p),0);
+	
+	
+}
+
+TEST(testCustomDensity, SimpleTest) {
+	
+	CustomDensity MD;
+	
+	Vector3d p(50*pc, 20*pc, -100*pc);	//random position for testing density 
+	
+	//try to get density without load any option in. Should give Warning in log-File and return density of 0.
+	EXPECT_DOUBLE_EQ(MD.getDensity(p),0);	
+	
+	MD.add(new ConstantDensity(1,1,1));	//all types loaded
+	MD.add(new ConstantDensity(0,2,0));	//overwrite HII type 
+	MD.add(new ConstantDensity(0,0,3));	//overwrite H2 type
+
+
+	
+	//check get density output
+	EXPECT_DOUBLE_EQ(MD.getHIDensity(p),1);
+	EXPECT_DOUBLE_EQ(MD.getHIIDensity(p),2);
+	EXPECT_DOUBLE_EQ(MD.getH2Density(p),3);
+	EXPECT_DOUBLE_EQ(MD.getDensity(p),6);	//total density 2+2+3
+	EXPECT_DOUBLE_EQ(MD.getNucleonDensity(p),9);	// nucleon density = 2 + 2 + 2*3  factor 2 for molecular hydrogen
+	
+	//check get function for type 
+	EXPECT_TRUE(MD.getisforHI());
+	EXPECT_TRUE(MD.getisforHII());
+	EXPECT_TRUE(MD.getisforH2());
+	
+	//check type deactivate funktion
+	MD.setisforHI(false);
+	EXPECT_FALSE(MD.getisforHI());
+	EXPECT_TRUE(MD.getisforHII());
+	EXPECT_TRUE(MD.getisforH2());
+	
+	MD.setisforHII(false);
+	EXPECT_FALSE(MD.getisforHI());
+	EXPECT_FALSE(MD.getisforHII());
+	EXPECT_TRUE(MD.getisforH2());
+	
+	MD.setisforH2(false);
+	EXPECT_FALSE(MD.getisforHI());
+	EXPECT_FALSE(MD.getisforHII());
+	EXPECT_FALSE(MD.getisforH2());
+	
+	//check density output if all types are deactivated
+	//should give error message in log-file and return density of 0.
+	EXPECT_DOUBLE_EQ(MD.getDensity(p),0.);
+	EXPECT_DOUBLE_EQ(MD.getNucleonDensity(p),0.);
+
+} 
+
+TEST(testDensityList, SimpleTest) {
+
+	DensityList MS;
+	MS.addDensity(new ConstantDensity(1,1,2));	//sum 4
+	MS.addDensity(new ConstantDensity(2,3,1));	//sum 6
+	
+	Vector3d p(50*pc,10*pc,-30*pc);	//random position for testing density
+	EXPECT_DOUBLE_EQ(MS.getHIDensity(p),3);
+	EXPECT_DOUBLE_EQ(MS.getHIIDensity(p),4);
+	EXPECT_DOUBLE_EQ(MS.getH2Density(p),3);
+	EXPECT_DOUBLE_EQ(MS.getDensity(p),10);	//sum of sums
+	EXPECT_DOUBLE_EQ(MS.getNucleonDensity(p),13); 	// 3+4+2*3 factor 2 for molecular hydrogen
+	
+}
+
+TEST(testCordes, SimpleTest) {
+
+	Cordes n;
+	
+	//check type Information
+	EXPECT_FALSE(n.getisforHI());
+	EXPECT_TRUE(n.getisforHII());
+	EXPECT_FALSE(n.getisforH2());
+
+	Vector3d p(3.1*kpc,2.9*kpc,-30*pc);	//position for testing density	
+	
+	EXPECT_NEAR(n.getHIIDensity(p), 184500.,1);	// output in m^-3 ; uncertainty of 1e-6 cm^-3 
+	EXPECT_NEAR(n.getDensity(p), 184500.,1);	
+	EXPECT_NEAR(n.getNucleonDensity(p), 184500,1);	// only HII component -> no differenz between density and nucleon density
+	p.z=30*pc;			// invariant density for +/- z
+	EXPECT_NEAR(n.getDensity(p),184500,1); 
+	
+}
+
+TEST(testNakanishi, SimpleTest) {
```

Comment:
SimpleTest is ot a very descriptive name - how about: checkValueAtCertainPonints.